frontend/workflows/projectSelector/src/hello-world.tsx

```diff
@@ -84,17 +84,24 @@ const useDispatch = () => {
 
 const selectorReducer = (state: State, action: Action): State => {
   switch (action.type) {
-    case "ADD_PROJECTS":
-      // TODO: don't add if it already exists.
+    case "ADD_PROJECTS": {
+      // a given custom project may already exist in the group so don't trigger a state update for those duplicates
+      const uniqueCustomProjects = action.payload.projects.filter(
+        (project: string) => !(project in state[action.payload.group])
+      );
```

Comment:
initially had this simple check https://github.com/lyft/clutch/pull/1507/commits/5152901f0bbb5c694b4abe48938c7052433ff03b#diff-0bcc1b146aa20a98ff28606102658d758469edad49c9a8f8ab6a9ed8fcf0e244R320-R323 but thought it might be better to move it here, especially if we plan to support bulk adding projects via the text box in the future.

Reply:
yep, that way if we add some other way to add projects in addition to the text box (e.g. query param) the logic still works. :+1: